server/src/main/java/com/google/fhir/proxy/HttpFhirClient.java

```diff
@@ -42,8 +42,44 @@ public abstract class HttpFhirClient {
   private static final Logger logger = LoggerFactory.getLogger(HttpFhirClient.class);
 
   // The list of header names to keep in a response sent from the proxy; use lower case only.
+  // Reference documentation - https://www.hl7.org/fhir/http.html#ops,
+  // https://www.hl7.org/fhir/async.html
   // Note we don't copy content-length/type because we may modify the response.
-  static final Set<String> HEADERS_TO_KEEP = Sets.newHashSet("last-modified", "date");
+  static final Set<String> RESPONSE_HEADERS_TO_KEEP =
+      Sets.newHashSet(
+          "last-modified",
+          "date",
+          "expires",
+          "content-location",
+          "etag",
+          "location",
+          "x-progress",
+          "x-request-id",
+          "x-correlation-id");
+
+  // The list of incoming header names to keep for forwarding request to the FHIR server; use lower
+  // case only.
+  // Reference documentation - https://www.hl7.org/fhir/http.html#ops,
+  // https://www.hl7.org/fhir/async.html
+  // We should NOT copy Content-Length as this is automatically set by the RequestBuilder when
+  // setting content Entity; otherwise we will get a ClientProtocolException.
+  static final Set<String> REQUEST_HEADERS_TO_KEEP =
```

Comment:
Reason for not adding accept

I didn't add this because we are always setting the response content type to a default header - [proxyResponse.getResponseWriter(](https://github.com/google/fhir-access-proxy/blob/cd1e01e502b3aa4798f29f7e72f5909479417601/server/src/main/java/com/google/fhir/proxy/BearerAuthorizationInterceptor.java#L298)

IIUC we expect the FHIR response to be in JSON. If we send application/xml then this can break the code.
[fhir-access-proxy/plugins/src/main/java/com/google/fhir/proxy/plugin/AccessGrantedAndUpdateList.java](https://github.com/google/fhir-access-proxy/blob/cd1e01e502b3aa4798f29f7e72f5909479417601/plugins/src/main/java/com/google/fhir/proxy/plugin/AccessGrantedAndUpdateList.java#L74)

Line 74 in [cd1e01e](https://github.com/google/fhir-access-proxy/commit/cd1e01e502b3aa4798f29f7e72f5909479417601)

 IParser parser = fhirContext.newJsonParser(); 
I tried the create Patient API with "Accept: application/fhir+xml" but I get this error in AccessGrantedAndUpdateList "Content does not appear to be FHIR JSON, first non-whitespace character was: "

We need to do much more changes to make sure client provided Accept header works. I can raise a different issue for this but is outside the scope of this issue.
Please correct me if this is not right.

Reply:
I see, yes you are right; filing an issue is a great idea (and please consider adding a reference to that issue here and comment about why we are not passing `accept`).